Add Release notes on v2.0.0 (#751)
aws-greengrass/aws-greengrass-lite: RELEASE_NOTES.md

# Release Notes v2.0.0 (Dec 16 2024)

This the first release for Greengrass nucleus lite (GGLite) product line. GGLite
v2 aims to be compatable with AWS's GreenGrass v2(GGV2) product line, however

> currently only subset of the features are supported with this release.

Reviewer: ```suggestion
currently only a subset of the features are supported with this release.
```

## Installing from sources

To install GGLite from source, please follow the guide from
[SETUP.md](./docs/SETUP.md) and [TES.md](./docs/TES.md), once the setup

> environment is complete, please start refering to

Reviewer: ```suggestion
 is complete, please start refering to
```

Author: I want to keep the setup environment as this is in relation with setting up a environment or the follow up where a developer can setup a nix environment

> [INSTALL.md](./docs/INSTALL.md) file.

Reviewer: ```suggestion
[INSTALL.md](./docs/INSTALL.md).
```

> For developers planning to contribute to GGLite, it's highly recommended to
> setup a nix environment, more information on setup can be found
> [here](./docs/DEVELOPMENT.md).

For those intrested in a generic method of provisioning GGLite device, fleet
provisioning by claim may be one of the appropriate solution to the problem.
Please refer to fleet provisioning setup guide
[here](./docs/Fleet-provisioning.md).

## New on this release

AWS takes pride in open source projects and support from community in building
these projects. To provide more transparency on decisions and be in sync with
the community all the design docs and specs for the project are now provided
with this release. You can find more details on the [design doc](./docs/design/)
and [spec doc](./docs/spec/) by following the corresponding links.

> As mentioned at the start, currently GGLite isn't fully compatable with GGV2.

Reviewer: Is GGV2 the name of the Java greengrass library?

> GGV2. The plan is to release this initail version of nucleus lite with basic

Reviewer: GGV2 - is that the official name of the product?

> The plan is to release this initail version of GGLite with basic feature support

Reviewer: ```suggestion
The plan is to release this initial version of GGLite with basic feature support
```

feature support first and then gain full feature compatibility with future

> releases. Currently only basic recipe types are supported and inforation on the

Reviewer: ```suggestion
releases. Currently only basic recipe types are supported and information on the
```

supported recipe can be found here [here](./docs/RECIPE_SUPPORT_CHANGES.md).

> Furthermore there are known difference in behavior that are outlined by

Reviewer: ```suggestion
Furthermore there are known differences in behavior that are outlined by
```

[KNOWN_ISSUES](./docs/KNOWN_ISSUES.md) document. Also please look at the Github

> issues section to keep up with latest known issues. It's encouraged to open up a

Reviewer: ```suggestion
issues section to keep up with latest known issues. Please create a
```
